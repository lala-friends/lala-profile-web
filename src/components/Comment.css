.comment {
    border: 1px solid #f5f5f5;
    padding: 16px;
    text-align: left;
    margin-bottom: 0.5em;
}

.flexbox {
    justify-content: flex-start;
}

.comment .date {
    color: #868e96;
}

.comment .flexbox div {
    margin-right: 1em;
}