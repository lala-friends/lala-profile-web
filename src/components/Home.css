.home {
    height: 100%;
    margin-top: -8em;
}

.home .title {
    margin-bottom: 1.5em;
    font-weight: 600;
    color: #495057;
    font-size: 36px;
    line-height: 1.2;
    text-align: left;
}

.home .sub {
    font-size: 18px;
    font-weight: 400;
    color: #868e96;
    line-height: 1.5;
    text-align: left;
}

.elem1 {
    height: 70%;
    padding-top: 15em;
}

.elem2 {
    height: 30%;
    background: #e3fafc;
}