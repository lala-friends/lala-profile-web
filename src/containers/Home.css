.home {
    height: 100%;
    margin-top: -8em;
}

.home .title {
    margin-bottom: 1.5em;
    font-weight: 600;
    color: #495057;
    font-size: 36px;
    line-height: 1.2;
    text-align: left;
}

.home .sub {
    font-size: 18px;
    font-weight: 400;
    color: #868e96;
    line-height: 1.5;
    text-align: left;
}

.elem1 {
    padding-top: 15em;
    padding-bottom: 10em;
}

.elem2 {
    background: #e3fafc;
}