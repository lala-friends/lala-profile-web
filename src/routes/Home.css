.home {
    background-color: #ffdeeb;
}
.home .container {
    margin-top: 8em;
    padding-top: 3em;
    padding-bottom: 3em;
}
.home .title {
    margin-bottom: 1.5em;
    font-weight: 600;
    color: #495057;
    font-size: 36px;
    line-height: 1.2;
}

.home .sub {
    font-size: 18px;
    font-weight: 400;
    color: #868e96;
    line-height: 1.5;
}