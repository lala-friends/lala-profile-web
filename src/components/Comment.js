import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './Comment.css';
import {dateYYYYMMDDHHMMSS} from '../utils/DateUtil';

class Comment extends Component {
    render() {
        return (
            <div className="comment">
                <div className="flexbox">
                    <div id="email" className="email">{this.props.email}</div>
                    <div id="date" className="date">{dateYYYYMMDDHHMMSS(this.props.date)}</div>
                </div>
                <div id="comment">{this.props.comment}</div>
            </div>
        )
    } 
}

Comment.propTypes = {
    email: PropTypes.string.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.instanceOf(Date)
};

export default Comment;