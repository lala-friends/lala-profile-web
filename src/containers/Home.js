import React from 'react';
import './Home.css';

const Home = () => (
    <div className="home container">
        <div className='text-group'>
            <div className="title">LALA FRIENDS are doing something FUN!</div>
            <div className="sub">Do you have any idea? Do you have any inconvenience?</div>
            <div className="sub">Let's talk together.</div>
            <div className="sub">So... let's get started.</div>
        </div>
    </div>
);


export default Home;