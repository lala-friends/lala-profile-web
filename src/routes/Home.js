import React from 'react';
import './Home.css';

const Home = () => {
    return (
        <div className="home">
            <div className="container">
                <div className="title">LALA FRIENDS are doing something FUN!</div>
                <div className="sub">Do you have any idea? Do you have any inconvenience?</div>
                <div className="sub">Let's talk together.</div>
                <div className="sub">So... let's get started.</div>
            </div>
        </div>
    )
}

export default Home;