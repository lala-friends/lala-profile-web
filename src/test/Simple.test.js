import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import Comment from '../components/Comment';
import Home from '../components/Home';
import NoMatch from '../components/NoMatch';

describe('단순한 컴포넌트의 테스트', () => {
    test('Comment가 렌더링 된다.', () => {
        const wrapper = shallow(<Comment email='tiffany@gmail.com' comment='comment 테스트' date={new Date('2017-01-01 00:00:00')}/>);
        expect(wrapper.find('#email').text()).toEqual('tiffany@gmail.com');
        expect(wrapper.find('#comment').text()).toEqual('comment 테스트');
        expect(wrapper.find('#date').text()).toEqual('2017-01-01 00:00:00');
    });

    test('Home이 렌더링 된다.', () => {
        const wrapper = shallow(<Home/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('NoMatch가 렌더링 된다.', () => {
       const wrapper = shallow(<NoMatch/>);
       expect(wrapper).toMatchSnapshot();
    });
});