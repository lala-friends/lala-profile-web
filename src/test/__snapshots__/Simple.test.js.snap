// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`단순한 컴포넌트의 테스트 Home이 렌더링 된다. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="elem1"
>
          <div
                    className="container"
          >
                    <div
                              className="title"
                    >
                              LALA FRIENDS are doing something FUN!
                    </div>
                    <div
                              className="sub"
                    >
                              Do you have any idea? Do you have any inconvenience?
                    </div>
                    <div
                              className="sub"
                    >
                              Let's talk together.
                    </div>
                    <div
                              className="sub"
                    >
                              So... let's get started.
                    </div>
          </div>
</div>,
        <div
          className="elem2"
/>,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <div
                        className="title"
            >
                        LALA FRIENDS are doing something FUN!
            </div>
            <div
                        className="sub"
            >
                        Do you have any idea? Do you have any inconvenience?
            </div>
            <div
                        className="sub"
            >
                        Let's talk together.
            </div>
            <div
                        className="sub"
            >
                        So... let's get started.
            </div>
</div>,
          "className": "elem1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                LALA FRIENDS are doing something FUN!
</div>,
              <div
                className="sub"
>
                Do you have any idea? Do you have any inconvenience?
</div>,
              <div
                className="sub"
>
                Let's talk together.
</div>,
              <div
                className="sub"
>
                So... let's get started.
</div>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LALA FRIENDS are doing something FUN!",
                "className": "title",
              },
              "ref": null,
              "rendered": "LALA FRIENDS are doing something FUN!",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Do you have any idea? Do you have any inconvenience?",
                "className": "sub",
              },
              "ref": null,
              "rendered": "Do you have any idea? Do you have any inconvenience?",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Let's talk together.",
                "className": "sub",
              },
              "ref": null,
              "rendered": "Let's talk together.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "So... let's get started.",
                "className": "sub",
              },
              "ref": null,
              "rendered": "So... let's get started.",
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "elem2",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="elem1"
>
            <div
                        className="container"
            >
                        <div
                                    className="title"
                        >
                                    LALA FRIENDS are doing something FUN!
                        </div>
                        <div
                                    className="sub"
                        >
                                    Do you have any idea? Do you have any inconvenience?
                        </div>
                        <div
                                    className="sub"
                        >
                                    Let's talk together.
                        </div>
                        <div
                                    className="sub"
                        >
                                    So... let's get started.
                        </div>
            </div>
</div>,
          <div
            className="elem2"
/>,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <div
                            className="title"
              >
                            LALA FRIENDS are doing something FUN!
              </div>
              <div
                            className="sub"
              >
                            Do you have any idea? Do you have any inconvenience?
              </div>
              <div
                            className="sub"
              >
                            Let's talk together.
              </div>
              <div
                            className="sub"
              >
                            So... let's get started.
              </div>
</div>,
            "className": "elem1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  LALA FRIENDS are doing something FUN!
</div>,
                <div
                  className="sub"
>
                  Do you have any idea? Do you have any inconvenience?
</div>,
                <div
                  className="sub"
>
                  Let's talk together.
</div>,
                <div
                  className="sub"
>
                  So... let's get started.
</div>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LALA FRIENDS are doing something FUN!",
                  "className": "title",
                },
                "ref": null,
                "rendered": "LALA FRIENDS are doing something FUN!",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Do you have any idea? Do you have any inconvenience?",
                  "className": "sub",
                },
                "ref": null,
                "rendered": "Do you have any idea? Do you have any inconvenience?",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Let's talk together.",
                  "className": "sub",
                },
                "ref": null,
                "rendered": "Let's talk together.",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "So... let's get started.",
                  "className": "sub",
                },
                "ref": null,
                "rendered": "So... let's get started.",
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "elem2",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`단순한 컴포넌트의 테스트 NoMatch가 렌더링 된다. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoMatch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "NoMatch",
    },
    "ref": null,
    "rendered": "NoMatch",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "NoMatch",
      },
      "ref": null,
      "rendered": "NoMatch",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
